sage-ios/SAGE/SAGE/CheckinViewController.swift

```diff
@@ -0,0 +1,263 @@
+    let sessionTitleLabel = UILabel()
+    var startTime: NSTimeInterval = 0.0
+    var inSession: Bool = false
+
+    //
+    // MARK: - ViewController Lifecycle
+    //
+    override func loadView() {
+        self.view = self.checkinView
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.setupDefaultTitleLabel()
+        self.setupSessionTitleLabel()
+        self.setupBarButtonItems()
+        
+        self.checkinView.startButton.addTarget(self, action: "userPressedBeginSession:", forControlEvents: .TouchUpInside)
+        self.checkinView.endButton.addTarget(self, action: "userPressedEndSession:", forControlEvents: .TouchUpInside)
+
+        self.locationManager.delegate = self
+        self.locationManager.requestWhenInUseAuthorization()
+        // May wanna put in viewDidAppear() if we want to get the user's location every time the mapView appears
+        self.startGettingCurrentLocation()
+    }
+    
+    override func viewWillAppear(animated: Bool) {
+        // Change mode on loading view depending on whether user is
+        // mid-check in or not
+        if let storedStartTime = KeychainWrapper.stringForKey(KeychainConstants.kSessionStartTime) {
+            self.presentSessionMode(0)
+            self.inSession = true
+            self.startTime = NSTimeInterval(storedStartTime)!
+            self.updateSessionTime()
+        } else {
+            self.presentDefaultMode(0)
+            self.inSession = false
+        }
+    }
+    
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        UIApplication.sharedApplication().statusBarStyle = .LightContent
+    }
+    
+    //
+    // MARK: - Button event handling
+    //
+    @objc private func userPressedBeginSession(sender: UIButton!) {
+        if CLLocationManager.locationServicesEnabled() {
+            switch CLLocationManager.authorizationStatus() {
+            case .AuthorizedWhenInUse, .AuthorizedAlways:
+                //verify user's location first
+                // if verified: then do this
+                // if not: show a too far alert
+                if true {
+                    let alertController = UIAlertController(
+                        title: "Start mentoring session?",
+                        message: nil,
+                        preferredStyle: .ActionSheet)
+                    alertController.addAction(UIAlertAction(title: "Start Session", style: .Default, handler: { (action: UIAlertAction) -> Void in
+                        //save start time locally and start timer
+                        self.presentSessionMode(UIConstants.normalAnimationTime)
+                        self.inSession = true
+                        self.startTime = NSDate.timeIntervalSinceReferenceDate()
+                        self.updateSessionTime()
+                        KeychainWrapper.setString(String(format: "%f", self.startTime), forKey: KeychainConstants.kSessionStartTime)
```

Comment:
you can store objects in keychain through keychain wrapper - why not just store a date object (so you won't have to parse this later)